homeassistant/components/recorder/entity_registry.py

```diff
@@ -40,16 +39,11 @@ def entity_registry_changed_filter(
         """Handle entity_id changed filter."""
         return event_data["action"] == "update" and "old_entity_id" in event_data
 
-    @callback
-    def _setup_entity_registry_event_handler(hass: HomeAssistant) -> None:
-        """Subscribe to event registry events."""
-        hass.bus.async_listen(
-            er.EVENT_ENTITY_REGISTRY_UPDATED,
-            _async_entity_id_changed,
-            event_filter=entity_registry_changed_filter,
-        )
-
-    async_at_start(hass, _setup_entity_registry_event_handler)
```

Comment:
Why did we previously only register this at start?

Reply:
https://github.com/home-assistant/core/pull/79556

Reply:
From above:
> Note that `async_at_start` was added in #79556, however since these are queued, it seems fine to let the rename tasks run as soon as they come up in the queue as a rename could still happen before start as well

Reply:
OK, but before #79556 was merged, the listener was only set up if Home Assistant was already running.
Was that just a mistake to begin with?

Reply:
It wasn't a problem until we added `async_track_entity_registry_updated_event` and than everything it fired got ahead of the line